uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1001.php

```diff
@@ -164,10 +162,43 @@ private function up1001_civicrmActivityTypesList() {
    * @return array
    */
   private function up1001_civicrmLocationTypesList() {
-    return [
+    $locationsToDelete = [
       'Main',
-      'Other',
+      'Other'
     ];
+    $deleteableLocations = [];
+
+    $tableName = CRM_Core_BAO_LocationType::getTableName();
+    $references = CRM_Core_DAO::getReferencesToTable($tableName);
+
+    foreach ($locationsToDelete as $currentLocation) {
+      $deleteLocation = true;
+
+      foreach ($references as $currentReference) {
+        if (!is_a($currentReference, 'CRM_Core_Reference_Dynamic')) {
+          $refTable = $currentReference->getReferenceTable();
+          $refKey = $currentReference->getReferenceKey();
+
+          $q = "
+            SELECT *
+            FROM $refTable, $tableName
```

Comment:
to keep consistency with the rest of the code, please enclose the variables with `{}` (and no abbreviations!)